numpy/typing/_array_like.py

```diff
@@ -16,12 +18,15 @@
     else:
         HAVE_PROTOCOL = True
 
+_DType = TypeVar("_DType", bound="dtype[Any]")
+
 if TYPE_CHECKING or HAVE_PROTOCOL:
-    class _SupportsArray(Protocol):
-        @overload
```

Comment:
Turns out the protocol already works fine for both positional-only and positional-or-keyword arguments, even without these overloads.